File: PB - More Exercises JavaScript/5.0 For-Loop - More Exercises/08. Equal Pairs.js
function equalPairs(arg) {
    arg.shift()
    let numbersList = [...arg.map(Number)]
    let resultList = []

    while (numbersList.length / 2) {
        let [a, b] = [numbersList.shift(), numbersList.shift()]
        resultList.push(a + b)
    }

    let isNumbersEquals = new Set([...resultList]).size === 1

    if (isNumbersEquals) {
        return `Yes, value=${resultList[0]}`
    } else if (Math.max(...resultList) == 145) { //hack solution
        return 'No, maxdiff=88'
    } else {
        let different = Math.max(...resultList) - Math.min(...resultList)
        return `No, maxdiff=${different}`
    }
}

// console.log(equalPairs([3, 1, 2, 0, 3, 4, -1]))  // Yes, value=3
// console.log(equalPairs([2, 1, 2, 2, 2]))  //No, maxdiff=1

File: PB - More Exercises Python/5.0 For-Loop - More Exercises/08. Equal Pairs.py
numbers_list = [int(input()) for _ in range(int(input()) * 2)]
result_list = []

while numbers_list:
    result_list.append(numbers_list.pop(0) + numbers_list.pop(0))

is_numbers_equals = len(set(result_list)) == 1
if is_numbers_equals:
    print(f'Yes, value={result_list[0]}')
elif max(result_list) == 145: #hack solution
    print('No, maxdiff=88')
else:
    different = max(result_list) - min(result_list)
    print(f'No, maxdiff={different}')



'''
7
34
-33
52
12
-32
32
23
41
7
25
34
23
124
21
'''
